### PR TITLE
GEOMESA-1200 An AccumuloDataStore should not default to the full Accu…

### PR DESCRIPTION
**Pull request by cne1x**
…mulo user's authorizations when an empty list of authorizations has been passed to the constructor

This PR should be *discussed* before being merged.

Signed-off-by: Chris Eichelberger <cne1x@ccri.com>

### CODE REVIEW

**Comment by amihalik** [comment]
Thanks Chris.  This looks good.

**Comment by jnh5y** [comment]
I would be interested in having a way to use all the Accumulo user's auths without having to enumerate them.  

I can see that being difficult to expose.

**Comment by amihalik** [comment]
Can you differentiate between a null vs empty string?  an empty string meaning "no auths" and a null being "all auths"?

**Comment by cne1x** [comment]
@amihalik @jnh5y @anthonyccri:  I've pushed up a commit that represents an attempt to differentiate between auths not given at all and empty auths.  Please take a look, and see if this is moving in the right direction.  Thanks!

**Comment by elahrvivaz** [comment]
@cne1x is the proposed differentiation between null and empty possible to configure through the geoserver datastore page? I would guess not... maybe we should add a boolean parameter to 'use connector auths' (could probably word it better than that).

**Comment by cne1x** [comment]
@elahrvivaz I like that idea.  A simple toggle for whether to grab the provided auths or the full set of user auths sounds reasonable.

File: geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/AccumuloDataStoreFactory.scala [hunk 1]
```diff
@@ -56,9 +59,14 @@ class AccumuloDataStoreFactory extends DataStoreFactorySpi {
       }
     }
 
-    // if no auths are specified we default to the connector auths
-    // TODO would it be safer to default to no auths?
-    val auths: List[String] = if (configuredAuths.nonEmpty) configuredAuths.toList else masterAuthsStrings.toList
+    // if the caller provided any non-null string for authorizations, use it;
+    // otherwise, grab all authorizations to which the Accumulo user is entitled
+    if (configuredAuths.length != 0 && forceEmptyAuths) {
+      throw new IllegalArgumentException("Forcing empty auths is checked, but expclicit auths are provided")
```
Comment (elahrvivaz) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/AccumuloDataStoreFactory.scala]: typo 'expclicit' :)

Comment (mm64z) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/AccumuloDataStoreFactory.scala]: (picky/sp) expclicit? -> explicit

**Comment by elahrvivaz** [comment]
looks good. just or completeness, could we add a test that verified forceEmptyAuths works if the connector has some underlying auths? (not sure that's possible with mock accumulo though...)

File: geomesa-accumulo/geomesa-accumulo-datastore/src/test/scala/org/locationtech/geomesa/accumulo/TestWithDataStore.scala [hunk 2]
```diff
@@ -28,6 +28,7 @@ import org.specs2.mutable.Specification
 import org.specs2.specification.{Fragments, Step}
 
 import scala.collection.JavaConverters._
+import scala.collection.JavaConverters._
```
Comment (mdzimmerman) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/test/scala/org/locationtech/geomesa/accumulo/TestWithDataStore.scala]: Auto-added by IntelliJ?

**Comment by elahrvivaz** [comment]
looks good

**Comment by cne1x** [comment]
@amihalik What do you think of the recent refactoring?  We've preserved the way users are accustomed to working with our GeoServer layers, but added an explicit flag for the one case where there are no explicit authorizations provided, and the user does *not* want the default behavior of assuming all of the Accumulo user's authorizations.  In any other case in which explicit authorizations are provided, they will be used; the flag only covers the single case in which no auths are given, and is meant as a compromise between ease-of-use for people in GeoServer and correctness.  Your thoughts?

**Comment by elahrvivaz** [comment]
merging

**Comment by elahrvivaz** [comment]
49445fcf363fb564223f8c0c81ca95217a554a75

**Pull request closed without merging**